### PR TITLE
Part 6 of week 3, translations and charts API integration, small CSS update and Week 5 work

### PR DESCRIPTION
**Pull request by PaulLockett**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/CommentsDataServlet.java [hunk 1]
```diff
@@ -81,4 +73,21 @@ private String convertToJsonUsingGson(ArrayList<String> comments) {
...
   }
 
+  private void populateCommentsList(HttpServletRequest request){
+    comments.clear();
+
+    int maxComments = Integer.parseInt(request.getParameter("maxComments"));
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    
+    Query query = new Query("comment");
```
Comment (allyons-google) [inline portfolio/src/main/java/com/google/sps/servlets/CommentsDataServlet.java]: Can this be static / immutable?

File: portfolio/src/main/java/com/google/sps/servlets/CommentsDataServlet.java [hunk 2]
```diff
@@ -81,4 +73,21 @@ private String convertToJsonUsingGson(ArrayList<String> comments) {
     return json;
   }
 
+  private void populateCommentsList(HttpServletRequest request){
+    comments.clear();
+
+    int maxComments = Integer.parseInt(request.getParameter("maxComments"));
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```
Comment (allyons-google) [inline portfolio/src/main/java/com/google/sps/servlets/CommentsDataServlet.java]: Can this be static / immutable?

File: portfolio/src/main/java/com/google/sps/servlets/CommentsDataServlet.java [hunk 3]
```diff
@@ -81,4 +73,21 @@ private String convertToJsonUsingGson(ArrayList<String> comments) {
...
+    
+    Query query = new Query("comment");
+    PreparedQuery results = datastore.prepare(query);
+  
+    for (Entity entity : results.asIterable(FetchOptions.Builder.withLimit(maxComments))) {
+
+      String comment = (String) entity.getProperty("text");
```
Comment (allyons-google) [inline portfolio/src/main/java/com/google/sps/servlets/CommentsDataServlet.java]: I think we should perform type checks whenever casting

File: portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response)
+  throws IOException {
+
+    ArrayList<Entity> comments = getCommentsList(request);
+
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```
Comment (allyons-google) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java]: Can this be static / immutable?

File: portfolio/src/main/java/com/google/sps/servlets/CommentsDataServlet.java (hunk 2 above)
Comment (PaulLockett) [reply portfolio/src/main/java/com/google/sps/servlets/CommentsDataServlet.java]: yes

File: portfolio/src/main/java/com/google/sps/servlets/CommentsDataServlet.java (hunk 1 above)
Comment (PaulLockett) [reply portfolio/src/main/java/com/google/sps/servlets/CommentsDataServlet.java]: yes

File: portfolio/src/main/java/com/google/sps/servlets/CommentsDataServlet.java (hunk 3 above)
Comment (PaulLockett) [reply portfolio/src/main/java/com/google/sps/servlets/CommentsDataServlet.java]: I'm not sure what you mean by type checks? Should I be doing a try Catch? or a if statetment using instanceof?

File: portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java (hunk 4 above)
Comment (PaulLockett) [reply portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java]: yes

File: portfolio/src/main/java/com/google/sps/servlets/CommentsDataServlet.java (hunk 3 above)
Comment (allyons-google) [reply portfolio/src/main/java/com/google/sps/servlets/CommentsDataServlet.java]: I think instanceof would be ideal

Comment (PaulLockett) [reply portfolio/src/main/java/com/google/sps/servlets/CommentsDataServlet.java]: see most recent commit

**Review by allyons-google — APPROVED** [review]

**Pull request merged**